dev/doc/proof-engine.md

```diff
@@ -0,0 +1,130 @@
+Tutorial on the new proof engine for ML tactic writers
+======================================================
+
+Starting from Coq 8.5, a new proof engine has been introduced, replacing the old
+meta-based engine which had a lot of both expressivity and soundness drawbacks.
+The latter is deprecated and is slowly getting removed from the source code.
+
+The new engine relies on a monadic API defined in the `Proofview` module. Helper
+functions and higher-level operations are defined in the `Tacmach` and
+`Tacticals` modules, and end-user tactics are defined amongst other in the
+`Tactics` module.
+
+At the root of the engine is a representation of proofs as partial terms that
+can contain typed holes, called evars. An evar is essentially defined by its
```

Comment:
Using « evar » throughout the document is fine. But at least here, better remind that it means « existential variable ».